src/utilities/utility.ts

```diff
@@ -50,50 +51,75 @@ export function checkForErrors(execResults: IExecSyncResult[], warnIfError?: boo
+    const result = kubectl.getResource('namespace', namespaceName);
+    if (!result) {
+        core.debug(`Failed to get commits from cluster.`);
+        return '';
+    }
+    else {
+        if (!!result.stderr) {
+            core.debug(`${result.stderr}`);
+            return process.env.GITHUB_SHA;
+        }
+        else if (!!result.stdout) {
+            const annotationsSet = JSON.parse(result.stdout).metadata.annotations;
+            if (!!annotationsSet && !!annotationsSet[annotationKey]) {
+                return JSON.parse(annotationsSet[annotationKey].replace(/'/g, '"')).commit;
+            }
+            else {
+                return 'NA';
+            }
+        }
+    }
+}
+
+export async function getWorkflowFilePath(githubToken: string): Promise<string> {
+    let workflowFilePath = process.env.GITHUB_WORKFLOW;
+    if (!workflowFilePath.startsWith('.github/workflows/')) {
+        const githubClient = new GitHubClient(process.env.GITHUB_REPOSITORY, githubToken);
+        const response = await githubClient.getWorkflows();
+        if (response.statusCode == StatusCodes.OK
+            && !!response.body
+            && !!response.body.total_count) {
+            if (response.body.total_count > 0) {
+                for (let workflow of response.body.workflows) {
+                    if (process.env.GITHUB_WORKFLOW === workflow.name) {
+                        workflowFilePath = workflow.path;
+                        break;
+                    }
+                }
+            }
+        }
+        else if (response.statusCode != StatusCodes.OK) {
+            core.debug(`An error occured while getting list of workflows on the repo. Statuscode: ${response.statusCode}, StatusMessage: ${response.statusMessage}`);
+        }
+    }
+    return Promise.resolve(workflowFilePath);
+}
+
+export function annotateChildPods(kubectl: Kubectl, resourceType: string, resourceName: string, annotationKeyValStr: string, allPods): IExecSyncResult[] {
     const commandExecutionResults = [];
     let owner = resourceName;
     if (resourceType.toLowerCase().indexOf('deployment') > -1) {
         owner = kubectl.getNewReplicaSet(resourceName);
     }
 
-    if (allPods && allPods.items && allPods.items.length > 0) {
+    if (!!allPods && !!allPods.items && allPods.items.length > 0) {
         allPods.items.forEach((pod) => {
             const owners = pod.metadata.ownerReferences;
-            if (owners) {
+            if (!!owners) {
                 owners.forEach(ownerRef => {
                     if (ownerRef.name === owner) {
-                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, workflowAnnotations, true));
+                        commandExecutionResults.push(kubectl.annotate('pod', pod.metadata.name, [annotationKeyValStr], true));
                     }
                 });
             }
         });
     }
-
     return commandExecutionResults;
 }
 

```

Comment:
We're not annotating namespace anymore?

Reply:
we are still annotating namespace in `annotateResources` method in `deployment-helper`. It has been removed from here because the annotate namespace is no longer conditional.